lib/views/pages/home/content/ranking/ranking_card.dart

```diff
@@ -0,0 +1,133 @@
+import "package:flutter/material.dart";
+import "package:proxima/models/ui/ranking/ranking_element_details.dart";
+import "package:proxima/models/ui/user_avatar_details.dart";
+import "package:proxima/views/components/content/publication_header/user_profile_pop_up.dart";
+import "package:proxima/views/components/content/user_avatar/user_avatar.dart";
+
+/// A class that displays a ranking card.
+/// The card displays the user's rank, display name, and centauri points.
+/// The card is clickable and opens a user profile pop-up.
+/// The card's color can be customized.
+class RankingCard extends StatelessWidget {
+  const RankingCard({
+    super.key,
+    required this.rankingElementDetails,
+  });
+
+  final RankingElementDetails rankingElementDetails;
+
+  static const userRankrankTextKey = Key("userRankrankText");
+  static const userRankDisplayNameTextKey = Key("userRankDisplayNameText");
+  static const userRankCentauriPointsTextKey =
+      Key("userRankCentauriPointsText");
+  static const userRankAvatarKey = Key("userRankAvatar");
+
+  // Sized box for the rank text to make sure that the text is centered
+  // and that surrounding element spacing is consistent.
+  // This width is allow to display three digits.
+  static const _sizedBoxRankWidth = 50.0;
+
+  static const _rankOneColor = Color.fromARGB(255, 255, 218, 75);
+  static const _rankSecondColor = Color.fromARGB(255, 217, 218, 204);
+  static const _rankThirdColor = Color.fromARGB(255, 229, 153, 137);
+
+  // This text is used when the parameter [userRank] of the
+  // provided [RankingElementDetails] is null.
+  static const _nullRankText = "---";
+
+  void onTapUserPopUp(BuildContext context) {
```

Comment:
It is not good practice to pass the `BuildContext` arround like that. This should be a local function, directly defined inside `build`.